nakadi-producer/src/main/java/org/zalando/nakadiproducer/eventlog/impl/EventLog.java

```diff
@@ -26,4 +42,18 @@ public class EventLog {
     private String lockedBy;
     private Instant lockedUntil;
     private String compactionKey;
+    private UUID eid;
+
+    /**
+     * This is only needed for backward compatibility.
+     *
+     * <p>For instance 213 will be converted to "00000000-0000-0000-0000-0000000000d5"</p>
+     */
+    public UUID getEid() {
+        if (eid == null) {
+            return new UUID(0, id);
+        }
+
+        return eid;
```

Comment:
For performance reasons, we likely want to only do the conversion once.

```suggestion
        if (eid == null) {
            eid = new UUID(0, id);
        }

        return eid;
```

Currently we do it once before submitting, and once again after checking the failure results from Nakadi.
(It's been like that in the old code, but if we can improve this now, even better.)

This makes also the lookup a tiny bit faster, as it'll now be an == comparison before comparing the content.